### PR TITLE
[JUJU-4305] Inject ctrlConfigService in client/controller facade

### PR DESCRIPTION
**Pull request by anvial**
This PR replaces all legacy (state-based) ControllerConfig API calls with the ControllerConfig Domain calls in client/controller facade. Need to mention that this is the only place in the Juju codebase,
where we need to write Controller Config to DB.

*Why this change is needed and what it does.*

## Checklist

*If an item is not applicable, use `~strikethrough~`.*

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing

## QA steps

*Commands to run to verify that the change works.*

Unit-tests should pass:
```sh
go test github.com/juju/juju/apiserver/facades/client/controller/... -gocheck.v
```

Controller config should be in Dqlite DB:
```sh
juju bootstrap lxd lxd
juju switch controller
juju ssh 0
# in machine 0
apt install go-dqlite
dqlite -s 127.0.0.1:17666 controller
select * from controller_config
# Observ controller config data

```

### CODE REVIEW

File: apiserver/facades/client/controller/controller.go [hunk 1]
```diff
@@ -239,7 +249,7 @@ func (c *ControllerAPI) dashboardConnectionInfoForIAAS(
 		return nil, errors.Trace(err)
 	}
 	modelName := model.Name()
-	ctrCfg, err := c.state.ControllerConfig()
+	ctrCfg, err := c.ctrlConfigService.ControllerConfig(context.TODO())
```
Comment (SimonRichardson) [inline apiserver/facades/client/controller/controller.go:252]: Can we change this to `context.Background()`

File: apiserver/facades/client/controller/controller.go [hunk 2]
```diff
@@ -725,13 +736,18 @@ func (c *ControllerAPI) ConfigSet(args params.ControllerConfigSet) error {
 	if err := c.checkIsSuperUser(); err != nil {
 		return errors.Trace(err)
 	}
+	// Write Controller Config to Mongo.
 	if err := c.state.UpdateControllerConfig(args.Config, nil); err != nil {
 		return errors.Trace(err)
 	}
+	// Write Controller Config to DQLite.
+	if err := c.ctrlConfigService.UpdateControllerConfig(context.TODO(), args.Config, nil); err != nil {
```
Comment (SimonRichardson) [inline apiserver/facades/client/controller/controller.go:744]: And this one should be `context.Background()`

**Review by SimonRichardson — APPROVED** [review]

File: apiserver/facades/client/controller/controller.go [hunk 3]
```diff
@@ -725,13 +736,18 @@ func (c *ControllerAPI) ConfigSet(args params.ControllerConfigSet) error {
 	if err := c.checkIsSuperUser(); err != nil {
 		return errors.Trace(err)
 	}
+	// Write Controller Config to Mongo.
 	if err := c.state.UpdateControllerConfig(args.Config, nil); err != nil {
 		return errors.Trace(err)
 	}
+	// Write Controller Config to DQLite.
```
Comment (manadart) [inline apiserver/facades/client/controller/controller.go:743]: Lack of sentence case is jarring to read.

**Review by manadart — COMMENTED** [review]
I know it's landed, but some things to sort out.

**Pull request merged**